java/src/main/java/com/google/cloud/dataproc/templates/kafka/KafkaToGCS.java

```diff
@@ -0,0 +1,144 @@
+ * run in few different modes. In default mode kafka.gcs.output.mode is set to "append". This will
+ * write only the new row in streaming DataFrame/Dataset to write to sink. Other possible values for
+ * this property are: 1. Complete: All the rows in the streaming DataFrame/Dataset will be written
+ * to the sink every time there are some updates. 2. Update: Only the rows that were updated in the
+ * streaming DataFrame/Dataset will be written to the sink every time there are some updates. For
+ * detailed list of properties refer "KafkaToGCS Template properties" section in
+ * resources/template.properties file.
+ */
+public class KafkaToGCS implements BaseTemplate {
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaToGCS.class);
+  private String gcsOutputLocation;
+  private String gcsOutputFormat;
+  private String gcsCheckpointLocation;
+  private String kafkaBootstrapServers;
+  private String kafkaTopic;
+  private String kafkaMessageFormat;
+  private String kafkaSchemaUrl;
+  private String kafkaStartingOffsets;
+  private String kafkaOutputMode;
+  private Long kafkaAwaitTerminationTimeout;
+
+  public KafkaToGCS() {
+
+    gcsOutputLocation = getProperties().getProperty(KAFKA_GCS_OUTPUT_LOCATION);
+    gcsOutputFormat = getProperties().getProperty(KAFKA_GCS_OUTPUT_FORMAT);
+    kafkaBootstrapServers = getProperties().getProperty(KAFKA_BOOTSTRAP_SERVERS);
+    kafkaTopic = getProperties().getProperty(KAFKA_TOPIC);
+    kafkaMessageFormat = getProperties().getProperty(KAFKA_MESSAGE_FORMAT);
+    kafkaSchemaUrl = getProperties().getProperty(KAFKA_SCHEMA_URL);
+    gcsCheckpointLocation = gcsOutputLocation.concat("/checkpoint/");
+    kafkaStartingOffsets = getProperties().getProperty(KAFKA_STARTING_OFFSET);
+    kafkaOutputMode = getProperties().getProperty(KAFKA_GCS_OUTPUT_MODE);
+    kafkaAwaitTerminationTimeout =
+        Long.valueOf(getProperties().getProperty(KAFKA_GCS_AWAIT_TERMINATION_TIMEOUT));
+  }
+
+  @Override
+  public void runTemplate() throws TimeoutException, StreamingQueryException {
+    validateInputs();
+
+      // Initialize Spark session
+      SparkSession spark = SparkSession.builder().appName("Spark KafkaToGCS Job").getOrCreate();
+
+      KafkaReader reader = new KafkaReader();
+
+      LOGGER.info("Calling Kafka Reader");
+
+      // Reading Kafka stream into dataframe
+      Dataset<Row> processedData = reader.readKafkaTopic(spark, getProperties());
+
+      // Write the output to GCS location
+      processedData
+          .writeStream()
+          .format(gcsOutputFormat)
+          .outputMode(kafkaOutputMode)
+          .option("checkpointLocation", gcsCheckpointLocation)
+          .option("path", gcsOutputLocation)
+          .start()
+          .awaitTermination(kafkaAwaitTerminationTimeout);
+
+      LOGGER.info("KakfaToGCS job completed.");
+      spark.stop();
+  }
+
+
+  void validateInputs(){
+    if (StringUtils.isAllBlank(gcsOutputLocation)
```

Comment:
We are trying to improve codebase by using jakarta validatations (eg: JDBCToGCSConfig.java)
You can possibly make a followup github issue for it, if you want.

Reply:
Will raise a follow up github issue to include jakarta validations